linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/DiagnosticTracer.scala

```diff
@@ -48,6 +57,10 @@ class DiagnosticTracer(
     )
 
     routerCtxF.map { ctx =>
+      if (resp.isChunked) {
+        resp.setChunked(false)
```

Comment:
do you also need to do `resp.reader.discard()`?

Reply:
Yea, you are right! I got mixed up with the description of `resp.setChunked(...)`

```scala
/**
   * Manipulate the `Message` content mode.
   *
   * If `chunked` is `true`, any existing content will be discarded and further attempts
   * to manipulate the synchronous content will result in an `IllegalStateException`.
   *
   * If `chunked` is `false`, the synchronous content methods will become available
   * and the `Reader`/`Writer` of the message will be ignored by Finagle.
   */

I will add that in.
```

Reply:
based on that comment, does the reader.discard need to happen before setting chunked to false.  Otherwise the reader.discard will be ignored by finagle?